src/ansys/simai/core/utils/configuration.py

```diff
@@ -36,6 +45,13 @@ def prompt_for_input_factory(*args, **kwargs):
     return lambda: prompt_for_input(*args, **kwargs)
 
 
+def error_or_prompt(interactive_mode, **kwargs):
```

Comment:
nit: Not sure about the name here but it's not easy to find a good one.
Maybe `prompt_if_interactive` ? 

Method could also maybe be merged with `prompt_for_input` where `name` is a normal arg which would avoid having a mandatory kwarg.

Finally you could just name the argument `interactive` to make everything slightly less verbose.

Reply:
Yeah, let's rename the function to `prompt_if_interactive` and rename its argument to interactive :)

However, I'm not sure about merging it with `prompt_for_input`. `prompt_for_input` is well defined for requesting inputs, while `prompt_if_interactive` is pydantic and configuration specific.